src/logger.cpp

```diff
@@ -44,28 +48,106 @@ Logger::~Logger()
     delete mPimpl;
 }
 
-void Logger::removeAllErrors()
+size_t Logger::errorCount() const
+{
+    return mPimpl->mErrors.size();
+}
+
+IssuePtr Logger::error(size_t index) const
+{
+    IssuePtr issue = nullptr;
+    if ((index < mPimpl->mErrors.size()) && (mPimpl->mErrors.at(index) < mPimpl->mIssues.size())) {
+        issue = mPimpl->mIssues.at(mPimpl->mErrors.at(index));
+    }
+    return issue;
+}
+
+size_t Logger::warningCount() const
+{
+    return mPimpl->mWarnings.size();
+}
+
+IssuePtr Logger::warning(size_t index) const
+{
+    IssuePtr issue = nullptr;
+    if ((index < mPimpl->mWarnings.size()) && (mPimpl->mWarnings.at(index) < mPimpl->mIssues.size())) {
+        issue = mPimpl->mIssues.at(mPimpl->mWarnings.at(index));
+    }
+    return issue;
+}
+
+size_t Logger::hintCount() const
 {
+    return mPimpl->mHints.size();
+}
+
+IssuePtr Logger::hint(size_t index) const
+{
+    IssuePtr issue = nullptr;
+    if ((index < mPimpl->mHints.size()) && (mPimpl->mHints.at(index) < mPimpl->mIssues.size())) {
+        issue = mPimpl->mIssues.at(mPimpl->mHints.at(index));
+    }
+    return issue;
+}
+
+void Logger::removeAllIssues()
+{
+    mPimpl->mIssues.clear();
     mPimpl->mErrors.clear();
+    mPimpl->mWarnings.clear();
+    mPimpl->mHints.clear();
 }
 
-void Logger::addError(const ErrorPtr &error)
+void Logger::addIssue(const IssuePtr &issue)
 {
-    mPimpl->mErrors.push_back(error);
+    // When an issue is added
```

Comment:
Need a full stop at the end of the comment. Same below and maybe elsewhere in this PR.